source/utilities.cc

```diff
@@ -104,6 +105,43 @@ namespace aspect
       return static_cast<bool>(ifile); // only in c++11 you can convert to bool directly
     }
 
+    void
+    load_and_distribute_file_content(const std::string &filename,
+                                     const MPI_Comm &comm,
+                                     std::stringstream &datastream)
+    {
+      // Reset datastream
+      datastream.str(std::string());
+
+      if (Utilities::MPI::this_mpi_process(comm) == 0)
+        {
+          std::ifstream filestream(filename);
+          AssertThrow (filestream,
+                       ExcMessage (std::string("Could not open file <") + filename + ">."));
+
+          // Read data from disk
+          filestream >> datastream.rdbuf();
+          std::string data_string = datastream.str();
+          unsigned int filesize = data_string.size();
+
+          // Distribute data_size and data across processes
+          MPI_Bcast(&filesize,1,MPI_UNSIGNED,0,comm);
+          MPI_Bcast(&data_string[0],filesize,MPI_CHAR,0,comm);
```

Comment:
For another miniscule performance enhancement: You should be able to get away with only broadcasting the second part. At the receiving end, you will then not immediately know how large a buffer to provide, but you can get the size of the data packet you will receive using `MPI_Probe` that gives you an object that contains the size of the next transmission; use it to resize the buffer and then do the actual call to `MPI_Bcast`.

(At least I think that works with Bcast. It definitely works with MPI_Send and MPI_Recv, and there are a number of examples in the deal.II code base for you to look at.)